templates/epic.xml.jinja2

```diff
@@ -9,6 +9,44 @@
    without any other variables defined.
 -#}
 
+{# Absolute default configuration, defines features -#}
+{% if features is not defined -%}
+{%   set features = ( {
```

Comment:
This only covers the case when template is instantiated by an empty file? Might as well keep the full feature file as a reference instead.

Reply:
Not sure I understand what you mean.

Reply:
It is indeed what we get for an empty yaml file, which is what we produce as the default `epic.xml` that most checks are run against (i.e. the reference design without the alternatives). We are already hitting the limits of GitHub actions, so we can't double all of the pipelines. Originally, we wanted the template to be simply a way to avoid having lots of nearly duplicated xml files. The idea was always that running without input would give the full detector in the current main reference design. I think that's probably something to maintain. It is possible that after arches and brycecanyon we will run more codenamed versions, but I think if we make a decision that we for sure will use the mRICH for example, then we'd update the default epic.xml to reflect that.

Not sure if that answers the second part of your comment. Maybe you are suggesting we should keep a `default.yml` file around to then parse into epic.xml, or treat`full.yml` like that? That's another option indeed, but means that the jinja template requires context whereas this was an attempt to make the template useful as a standalone entity.

Reply:
I guess, my main concern is that this gets rid of `features is not defined or ` but does nothing for `'foo' is in features['bar'] or `.

Reply:
Using `full.yml` might be better

Reply:
I think I understand the concern better now: this PR attempts to handle the absence of `features` more gracefully but it does not handle the absence of an interable as `features['bar']`.

I think that's ok, since the goal is to have a sensible default that does not have a suffix (which is the user contract for files in `configurations/`), not to handle malformed input configuration (which should rightfully fail). Therefore, I think it is ok to fail when getting `features: pid:` without specifying *what* is in `pid:`.

To silently accept an empty `features: pid:`, we could use `{% if 'pid' in features and features['pid'] is iterable %}`, but I feel that the intent of including `features: pid:` is likely to include all default PID.

In any case, I'm thinking about how to make this a bit easier to manage, more expressive, and less error prone.

Reply:
Yes, from you original motivation

> It means that the default configuration (that gives us `epic.xml`) is defined in an entirely different way than the configuration files.

The approach from this PR will only further the discrepancy, by making the `epic.xml` into an one-off case.

Reply:
So, you're proposing to get rid of epic.xml entirely?

Reply:
And then we'd only have named configurations left.

Reply:
Removed defaults in jinja; using just `full.yml` as default which installs as `epic.xml`.